src/thermo/PengRobinson.cpp

```diff
@@ -0,0 +1,955 @@
+{
+
+const double PengRobinson::omega_a = 4.5723552892138218E-01;
+const double PengRobinson::omega_b = 7.77960739038885E-02;
+const double PengRobinson::omega_vc = 3.07401308698703833E-01;
+
+PengRobinson::PengRobinson() :
+    m_formTempParam(0),
+    m_b_current(0.0),
+    m_a_current(0.0),
+    m_aAlpha_current(0.0),
+    NSolns_(0),
+    dpdV_(0.0),
+    dpdT_(0.0)
+{
+    fill_n(Vroot_, 3, 0.0);
+}
+
+PengRobinson::PengRobinson(const std::string& infile, const std::string& id_) :
+    m_formTempParam(0),
+    m_b_current(0.0),
+    m_a_current(0.0),
+    m_aAlpha_current(0.0),
+    NSolns_(0),
+    dpdV_(0.0),
+    dpdT_(0.0)
+{
+    fill_n(Vroot_, 3, 0.0);
+    initThermoFile(infile, id_);
+}
+
+void PengRobinson::calculateAlpha(const std::string& species, double a, double b, double w)
+{
+    size_t k = speciesIndex(species);
+    if (k == npos) {
+        throw CanteraError("PengRobinson::calculateAlpha",
+            "Unknown species '{}'.", species);
+    }
+
+    // Calculate value of kappa (independent of temperature)
+    // w is an acentric factor of species and must be specified in the CTI file
+    if (isnan(w)){
+        throw CanteraError("PengRobinson::calculateAlpha",
+            "No acentric factor loaded.");
+    } else if (w <= 0.491) {
+        kappa_vec_[k] = 0.37464 + 1.54226*w - 0.26992*w*w;
+    } else {
+        kappa_vec_[k] = 0.374642 + 1.487503*w - 0.164423*w*w + 0.016666*w*w*w;
+    }
+
+    //Calculate alpha (temperature dependent interaction parameter)
+    double critTemp = speciesCritTemperature(a, b); // critical temperature of individual species
+    double sqt_T_r = sqrt(temperature() / critTemp);
+    double sqt_alpha = 1 + kappa_vec_[k] * (1 - sqt_T_r);
+    alpha_vec_Curr_[k] = sqt_alpha*sqt_alpha;
+}
+
+void PengRobinson::setSpeciesCoeffs(const std::string& species,
+                                        double a, double b, double w)
+{
+    size_t k = speciesIndex(species);
+    if (k == npos) {
+        throw CanteraError("PengRobinson::setSpeciesCoeffs",
+            "Unknown species '{}'.", species);
+    }
+    size_t counter = k + m_kk * k;
+    a_coeff_vec(0, counter) = a;
+    // we store this locally because it is used below to calculate a_Alpha:
+    double aAlpha_k = a*alpha_vec_Curr_[k];
+    aAlpha_coeff_vec(0, counter) = aAlpha_k;
+
+    // standard mixing rule for cross-species interaction term
+    for (size_t j = 0; j < m_kk; j++) {
+        if (k == j) {
+            continue;
+        }
+        double a0kj = sqrt(a_coeff_vec(0, j + m_kk * j) * a);
+        double aAlpha_j = a*alpha_vec_Curr_[j];
+        double a_Alpha = sqrt(aAlpha_j*aAlpha_k);
+        if (a_coeff_vec(0, j + m_kk * k) == 0) {
+            a_coeff_vec(0, j + m_kk * k) = a0kj;
+            aAlpha_coeff_vec(0, j + m_kk * k) = a_Alpha;
+            a_coeff_vec(0, k + m_kk * j) = a0kj;
+            aAlpha_coeff_vec(0, k + m_kk * j) = a_Alpha;
+        }
+    }
+    a_coeff_vec.getRow(0, a_vec_Curr_.data());
```

Comment:
I'm confused by the use of both `a_coeff_vec` and `a_vec_Curr_`. In the P-R model, `a` is just a constant for a given species pair, right? All of the temperature dependence comes in through alpha.

Reply:
You are right. Both are same in case of Peng-Robinson. I have removed a_coeff_vec to avoid the duplication.